File: tests/fixtures/sample.ts
// Sample TypeScript file with linting issues for testing

interface User {
    name:string;
    age:number;
}

function badFunction(x:number,y:number):number {
    return x+y;
}

// Unused variable
const unused = "this is unused";

// Missing spaces
const value=1+2;

// Semicolons missing (if configured)
const text = "hello"

// Any type usage
const data: any = { foo: "bar" };

// Unused function parameters
function process(user:User, id:string) {
    console.log(user.name);
}

File: tests/fixtures/sample.py
"""Sample Python file with linting issues for testing."""

import os
import sys


def bad_function( x,y,z ):
    """This function has formatting issues."""
    result=x+y+z
    return result


# Unused variable
unused = "this is unused"


# Missing whitespace around operators
value=1+2


# Line too long
very_long_variable_name_that_definitely_violates_common_style_guides = "hello world"


class BadClass:
    def __init__(self):
        self.value=1

    def bad_method(self):
        pass
